hydra-node/src/Hydra/Chain/Direct/Tx.hs

```diff
@@ -542,21 +542,18 @@ data InitObservation = InitObservation
 -- only returning a Maybe, i.e. 'Either Reason (OnChainTx tx, OnChainHeadState)'
 observeInitTx ::
   NetworkId ->
-  [VerificationKey PaymentKey] ->
   Party ->
   Tx ->
   Maybe InitObservation
-observeInitTx networkId cardanoKeys party tx = do
+observeInitTx networkId party tx = do
   -- FIXME: This is affected by "same structure datum attacks", we should be
   -- using the Head script address instead.
   (ix, headOut, headData, Head.Initial cp ps) <- findFirst headOutput indexedOutputs
   parties <- mapM partyFromChain ps
   let contestationPeriod = fromChain cp
   guard $ party `elem` parties
   (headTokenPolicyId, headAssetName) <- findHeadAssetId headOut
-  let expectedNames = assetNameFromVerificationKey <$> cardanoKeys
-  let actualNames = assetNames headAssetName
-  guard $ sort expectedNames == sort actualNames
+  guard $ headAssetName `elem` mintedAssetNames
```

Comment:
The `headAssetName` is the state token (ST). We should be looking for a participation token (PT). That is, the public key hash of our own cardano key (using `assetNameFromVerificationKey` as before)